service/src/test/java/org/mskcc/smile/service/TempoServiceTest.java

```diff
@@ -402,6 +402,22 @@ public void testFindSamplesNoLongerEmbargoed() throws Exception {
         Assertions.assertEquals(1, tempoIds.size());
     }
 
+    @Test
+    public void testInitAndSaveDefaultTempoDataWithFallbackToCohortDate() throws Exception {
+        // use a tumor sample that doet not belong to an existing cohort
+        String igoId = "MOCKREQUEST1_B_3";
+        Tempo preTempo = tempoService.getTempoDataBySamplePrimaryId(igoId);
+        Assertions.assertNull(preTempo);
+
+        // call initAndSaveDefaultTempoData with a fallback date
+        String fallbackCohortDate = "2024-06-01 12:00";
```

Comment:
maybe add ` tempoService.saveTempo(...)` call on this `preTempo` event. 

and then assert that the initial pipeline run date and embargo date are both null 

then the rest of the test leave as is so we can check the dates after the sample tempo data is saved with the fallback date